lib/private/Security/Normalizer/IpAddress.php

```diff
@@ -103,6 +103,8 @@ public function getSubnet(): string {
 				$this->ip,
 				32
 			);
+		} else if (substr(strtolower($this->ip), 0, 7) === '::ffff:') {
```

Comment:
```suggestion
		} elseif (substr(strtolower($this->ip), 0, 7) === '::ffff:') {
```

Does this always mean there is an IPv4 inside? Or can it also be another ipv6?

Reply:
I found this in https://en.wikipedia.org/wiki/IPv6_address#Transition_from_IPv4 but the text doesn't fully match
also https://en.wikipedia.org/wiki/IPv6_address#Special_addresses

I can't say for sure but the more I read about this the more confused I am...

Reply:
Yes, last 32 bits of the ipv6 address are considered to be interpreted as IPv4 (if prefix matches ::ffff:)

Basically the whole IPv4 adress space is then mapped into `::ffff:0:0/96`. The thing I'm not sure about here is, if expanded notation as in `0000:0000:0000:0000:0000:ffff:7f00:0001` should also taken into consideration, which would be the same as `::ffff:127.0.0.1`

Reply:
If I understand correctly the code you cannot know for sure the IP will be in canonical form yet?
How does the test behave if you add something like ['::0:ffff:192.168.0.123','::ffff:192.168.0.123/32'] ?

Otherwise from what I understand this prefix is reserved to IPv4 mapped addresses.

Reply:
yeah, probably I'll use the regexp instead and limit it strictly to "::ffff:x.x.x.x" format ?
or how many more formats do we need to add there ?

Reply:
(also wondering why we're not using a library instead of writing own code like this)

Reply:
I've now used the binary form to detect all these, see last commit